praw/models/preferences.py

```diff
@@ -0,0 +1,185 @@
+"""Provide the Preferences class."""
+from json import dumps
+
+from ..const import API_PATH
+
+
+class Preferences(object):
+    """A class for Reddit preferences.
+
+    The Preferences class provides access to the Reddit preferences of the
+    currently logged in user.
+    """
+
+    def __init__(self, reddit):
+        """Create a Preferences instance.
+
+        :param reddit: The Reddit instance.
+        """
+        self._reddit = reddit
+
+    def __call__(self):
+        """Return the preference settings of the logged in user as a dict.
+
+        This is intended to be accessed as ``reddit.user.preferences()``
+        like so:
+
+        .. code-block:: python
+
+           prefs = reddit.user.preferences()
+           print(prefs['show_link_flair'])
+
+        See https://www.reddit.com/dev/api#GET_api_v1_me_prefs for the list
+        of possible values.
+        """
+        return self._reddit.get(API_PATH['prefs'])
+
+    def update(self, **prefs):
+        """Modify the specified settings.
+
+        :param 3rd_party_data_personalized_ads: Allow Reddit to use data
```

Comment:
Python doesn't support variables names that start with numbers. Perhaps we should have an example for how to pass such values successfully since I don't think they can be called by keyword argument (but worth investigating).

Reply:
I mentioned this in the PR body, but you're right, I should put it in the docstring.

> Preferences that have names that would be illegal keyword-argument names in Python (there are three at present) can still be updated like so:
> 
> ```python
> reddit.user.preferences.update(**{'3rd_party_data_personalized_ads' = False})
> ```